_test/helpers.ts

```diff
@@ -16,6 +18,67 @@ export async function loadTargetFromReport(
 ) {
   const [, reportId] = /reports\/(\w+)$/.exec(url)!;
 
+  // Setup request interception
+  await page.route("**/**/v1/**", async (route) => {
+    const request = route.request();
+    const url = new URL(request.url());
+    const apiPath = url.pathname.replace("/v1/", "");
+    const searchParams = url.searchParams;
+
+    // Determine mock file path based on API endpoint
+    let mockFile: string;
+    if (apiPath.startsWith("report/fights/")) {
+      // Base report fights endpoint
+      mockFile = path.join("_test", "mock", "report", reportId, "fights.json");
+    } else if (apiPath.startsWith("report/events/")) {
+      // Events endpoint with filter
+      const start = searchParams.get("start");
+
+      if (!start) {
+        throw new Error("Missing 'start' parameter in events request");
+      }
+
+      mockFile = path.join(
+        "_test",
+        "mock",
+        "report",
+        reportId,
+        "fights",
+        start,
+        "events.json"
+      );
+    } else {
+      throw new Error(`Unhandled API endpoint: ${apiPath}`);
+    }
+
+    try {
+      // Try to load existing mock data
+      const mockData = JSON.parse(fs.readFileSync(mockFile, "utf-8"));
+      await route.fulfill({ json: mockData });
+    } catch (error) {
+      // If mock file doesn't exist, let the request through and save the response
+      console.log(
+        `No mock data found for ${apiPath}, saving response to ${mockFile}`
+      );
+
+      // Create directory if it doesn't exist
+      const dir = path.dirname(mockFile);
+      if (!fs.existsSync(dir)) {
```

Comment:
Here if we don't find the mock data, it will read the api response and write a new mock file. So mocking is very easy.

If we need to change theevents that come back (show, hide some ability), then we can delete all mocks and run tests and it will regenerate all mocks for us.